Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -391,11 +401,43 @@ gdjs.LightRuntimeObjectPixiRenderer.prototype._computeLightVertices = function (
       obstaclePolygons.push(obstacleHitBoxes[i][j]);
   }
 
+  var maxX = this._object.x + this._radius;
+  var minX = this._object.x - this._radius;
+  var maxY = this._object.y + this._radius;
+  var minY = this._object.y - this._radius;
+
   var flattenVertices = [];
-  for (var i = 0; i < obstaclePolygons.length; i++) {
+  for (var i = 1; i < obstaclePolygons.length; i++) {
     var vertices = obstaclePolygons[i].vertices;
     var verticesCount = vertices.length;
-    for (var j = 0; j < verticesCount; j++) flattenVertices.push(vertices[j]);
+    for (var j = 0; j < verticesCount; j++) {
+      flattenVertices.push(vertices[j]);
+
+      if (vertices[j][0] < minX) minX = vertices[j][0];
+      if (vertices[j][0] > maxX) maxX = vertices[j][0];
+      if (vertices[j][1] < minY) minY = vertices[j][1];
+      if (vertices[j][1] > maxY) maxY = vertices[j][1];
+    }
+  }
+
+  obstaclePolygons[0].vertices[0][0] = minX;
+  obstaclePolygons[0].vertices[0][1] = minY;
+  obstaclePolygons[0].vertices[1][0] = maxX;
+  obstaclePolygons[0].vertices[1][1] = minY;
+  obstaclePolygons[0].vertices[2][0] = maxX;
+  obstaclePolygons[0].vertices[2][1] = maxY;
+  obstaclePolygons[0].vertices[3][0] = minX;
+  obstaclePolygons[0].vertices[3][1] = maxY;
+
+  var halfOfDiag = Math.max(
+    Math.hypot(this._object.x - minX, this._object.y - minY),
+    Math.hypot(maxX - this._object.x, this._object.y - minY),
+    Math.hypot(maxX - this._object.x, maxY - this._object.y),
+    Math.hypot(this._object.x - minX, maxY - this._object.y)
+  )
```

Comment:
Now, since the expansion of the bounding box is not equal (ie, maybe only the `minX` changes and the bbox becomes larger from the left side but the right size is still equal to what it was before), so the center of the light is not necessarily the center of the rectangle. So we find the distance of all 4 vertices from the center of the light (also, I now realize that they are not really the diagonals) and take the maximum, which is then passed to `_computeClosestIntersectionPoint`. Maybe we should name this something like `maxLengthToBboxVertices` and not `halfOfDiag` since they aren't diagonals?

Reply:
Yep we should rename this variable. We can still consider this to be some half diagonal value of an hypothetic bounding square, so maybe `boundingSquareHalfDiagonal`?